zipline/assets/assets.py

```diff
@@ -922,8 +922,9 @@ def get_ordered_contracts(self, root_symbol):
 
     def create_continuous_future(self, root_symbol, offset, roll_style):
         oc = self.get_ordered_contracts(root_symbol)
-        start_date = self.retrieve_asset(oc.contract_sids[0]).start_date
-        end_date = self.retrieve_asset(oc.contract_sids[-1]).end_date
+        start_date = min(a.end_date for a in self.retrieve_all(
```

Comment:
Also, adding a test to cover this, since this shows a lack of coverage on my part.